### PR TITLE
Kernel & DynamicLoader: A couple of small fixes

### PR DESCRIPTION
**Pull request by IdanHo**
Just a couple of patches I had laying around and never actually PRed.

### CODE REVIEW

File: Kernel/Memory/MemoryManager.cpp [hunk 1]
```diff
@@ -711,16 +711,21 @@ PageFaultResponse MemoryManager::handle_page_fault(PageFault const& fault)
...
 
     if (faulted_in_range(&start_of_unmap_after_init, &end_of_unmap_after_init)) {
         auto const* kernel_symbol = symbolicate_kernel_address(fault.vaddr().get());
-        PANIC("Attempt to access UNMAP_AFTER_INIT section ({:p}: {})", fault.vaddr(), kernel_symbol ? kernel_symbol->name : "(Unknown)");
+        dbgln("Attempt to access UNMAP_AFTER_INIT section ({:p}: {})", fault.vaddr(), kernel_symbol ? kernel_symbol->name : "(Unknown)");
+        return PageFaultResponse::ShouldCrash;
     }
 
-    if (faulted_in_range(&start_of_kernel_ksyms, &end_of_kernel_ksyms))
-        PANIC("Attempt to access KSYMS section");
```
Comment (supercomputer7) [inline Kernel/Memory/MemoryManager.cpp:723]: Isn't this considered a kernel bug?

File: Kernel/Memory/MemoryManager.cpp [hunk 2]
```diff
@@ -711,16 +711,21 @@ PageFaultResponse MemoryManager::handle_page_fault(PageFault const& fault)
...
-        PANIC("Attempt to write into READONLY_AFTER_INIT section");
+    if (faulted_in_range(&start_of_ro_after_init, &end_of_ro_after_init)) {
+        dbgln("Attempt to write into READONLY_AFTER_INIT section");
+        return PageFaultResponse::ShouldCrash;
+    }
 
     if (faulted_in_range(&start_of_unmap_after_init, &end_of_unmap_after_init)) {
         auto const* kernel_symbol = symbolicate_kernel_address(fault.vaddr().get());
-        PANIC("Attempt to access UNMAP_AFTER_INIT section ({:p}: {})", fault.vaddr(), kernel_symbol ? kernel_symbol->name : "(Unknown)");
+        dbgln("Attempt to access UNMAP_AFTER_INIT section ({:p}: {})", fault.vaddr(), kernel_symbol ? kernel_symbol->name : "(Unknown)");
```
Comment (supercomputer7) [inline Kernel/Memory/MemoryManager.cpp:721]: Isn't this considered a kernel bug?

File: Kernel/Memory/MemoryManager.cpp (hunk 1 above)
Comment (IdanHo) [reply Kernel/Memory/MemoryManager.cpp:723]: No, not necessarially, there's no check here that we were in kernel mode, a process can try to access this just as well.
(and the caller of this function panics if this fault _was_ in kernel mode and the response was ShouldCrash)

Comment (supercomputer7) [reply Kernel/Memory/MemoryManager.cpp:723]: OK, seems fine then to me :)
Thanks for the explanation!

**Pull request merged**